src/util/request-utils.ts

```diff
@@ -526,14 +526,15 @@ type PartiallyParsedHttpRequest = {
 // very lax, and will throw errors due to unexpected response data, but it's used when we
 // ourselves generate the data (for websocket responses that 'ws' writes directly to the
 // socket invisibly). Fortunately all responses are very simple:
-export function parseRawHttpResponse(input: Buffer): CompletedResponse {
-    const res: Partial<CompletedResponse> = {};
+export function parseRawHttpResponse(input: Buffer, request: OngoingRequest): CompletedResponse {
```

Comment:
The `parseRawHttpResponse` now accepts the ongoing request. This allows this method to create a full-fledged `CompletedRequest` without having to use TypeScript type guards.